### DISCUSSION TITLE
Feature request: Add `fill: var(--fa-primary-color, currentcolor)` to all SVG elements and their inner paths/shapes for easy color casting when working with raw SVG code

### DISCUSSION THREAD

**Discussion by brandonmcconnell** [FortAwesome/Font-Awesome#20902] [Feature Requests]
### Feature description

It would help when working with raw SVG code from FontAwesome to add `fill: var(--fa-primary-color, currentcolor)` to all SVG paths so that by default, if no color is set on the icon, the icons match their context and the current color surrounding them rather than defaulting to black, and can also be overridden by the common FA CSS variable `--fa-primary-color`.

Notably, because this prioritizes `--fa-primary-color` over `currentcolor`, this will work for all FA implementations and not produce any breaking changes. Simply, this will fall back to `currentcolor` instead of `black` when `--fa-primary-color` is unset and use `--fa-primary-color` otherwise.

For example, when I pull in this URL via a `fetch` call:
https://ka-p.fontawesome.com/releases/v5.15.2/svgs/solid/scalpel.svg?token=XXXXXXXXXX

this is what is returned:
```html
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
  <!-- Font Awesome Pro 5.15.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) -->
  <path d="M482.7 11.85c-29.2-20.83-70.18-13.03-93.49 14.22l-201.5 235.46c-8.9 10.41-1.51 26.47 12.19 26.47h131.94c9.37 0 18.28-4.1 24.37-11.22l139.02-162.44c26.37-30.8 21.23-78.41-12.53-102.49zM0 512c87 .07 170.28-29.18 234.29-81.16.2-.16.39-.32.59-.48 31.37-25.71 46.72-63.93 46.72-102.32V320H176L0 512z"/>
</svg>
```

but this would be preferred:
```html
<svg style="fill: var(--fa-primary-color, currentcolor)" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
  <!-- Font Awesome Pro 5.15.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) -->
  <path style="fill: var(--fa-primary-color, currentcolor)" d="M482.7 11.85c-29.2-20.83-70.18-13.03-93.49 14.22l-201.5 235.46c-8.9 10.41-1.51 26.47 12.19 26.47h131.94c9.37 0 18.28-4.1 24.37-11.22l139.02-162.44c26.37-30.8 21.23-78.41-12.53-102.49zM0 512c87 .07 170.28-29.18 234.29-81.16.2-.16.39-.32.59-.48 31.37-25.71 46.72-63.93 46.72-102.32V320H176L0 512z"/>
</svg>
```

** Note that `style="fill: var(--fa-primary-color, currentcolor)"` was added to both the `svg` element as well as its `path` element here.

### Feature request checklist

- [X] The title starts with "Feature request: " and is followed by the requested feature description
- [X] This is a single feature
- [X] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome/issues) and to the best of my knowledge this is not a duplicate

**Comment by hudri** [comment]
Absolutely in support for this, this would be a huge QoL improvement. We are working in a non-JS SSR environment and are including the source of SVGs inline in our pages. The missing currentColor adds an unnecessary workload - every - single - time :-(

**Comment by StevenLantinga** [comment]
Weirdly enough at least `currentColor` was part of every svg in the past. But they seem to have stopped doing that.

**Comment by tagliala** [comment]
Hello, do you know if this is causing issues with CSP `style_src` when inline style is forbidden ?

**Comment by claudiobosticco** [comment]
this would also be very beneficial for sprites files

**Comment by Tragen** [comment]
I was looking for the same. Seems like it's not working anymore as the color is not there anymore.
If it's because of size, how about a second package with the option to change color?
